avro-builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/own/AvroUtilCodeGenOp.java

```diff
@@ -46,38 +49,42 @@ public AvroUtilCodeGenOp(CodeGenOpConfig config) {
 
   @Override
   public void run(OperationContext opContext) throws Exception {
-    //mkdir any output folders that dont exist
+    //mkdir any output folders that don't exist
     if (!config.getOutputSpecificRecordClassesRoot().exists() && !config.getOutputSpecificRecordClassesRoot()
         .mkdirs()) {
       throw new IllegalStateException(
           "unable to create destination folder " + config.getOutputSpecificRecordClassesRoot());
     }
 
-    List<Path> avscFiles = new ArrayList<>();
+    Set<File> avscFiles = new HashSet<>();
+    Set<File> importableFiles = new HashSet<>();
+    String[] extensions = new String[]{BuilderConsts.AVSC_EXTENSION};
+
+
     for (File inputRoot : config.getInputRoots()) {
-      Files.walk(inputRoot.toPath())
-          .filter(path -> path.getFileName().toString().endsWith("." + BuilderConsts.AVSC_EXTENSION))
-          .forEach(avscFiles::add);
+      avscFiles.addAll(FileUtils.listFiles(inputRoot, extensions, true));
+    }
+
+    if (config.getIncludeRoots() != null) {
+      for (File includeRoot : config.getIncludeRoots()) {
+        importableFiles.addAll(FileUtils.listFiles(includeRoot, extensions, true));
+      }
     }
-    if (avscFiles.isEmpty()) {
+
+    // If there are no importable files, then all of avscFiles should be marked as importable.
```

Comment:
i think this would result in unintuitive behaviour

Reply:
Good point, I'll adjust